toolkit/tools/internal/logger/log.go

```diff
@@ -245,3 +246,29 @@ func setHookLogLevel(hook *writerHook, level string) (err error) {
 	hook.SetLevel(logLevel)
 	return
 }
+
+// FormatWarningBox formats a message into a box with a border. The box is automatically sized to fit the longest line.
+// Each line will be centered in the box.
+func FormatWarningBox(message []string) []string {
+	var (
+		corners    = []string{"╔", "╗", "╚", "╝"}
+		horizontal = "═"
+		vertical   = "║"
+	)
+	maxLineLength := 0
+	for _, line := range message {
+		if len(line) > maxLineLength {
+			maxLineLength = len(line)
+		}
+	}
+	// Count: 2 for corners, 2 for padding, longest string
+	lines := []string{corners[0] + strings.Repeat(horizontal, maxLineLength+2) + corners[1]}
+	for _, line := range message {
+		paddingL := (maxLineLength - len(line)) / 2
+		paddingR := maxLineLength - len(line) - paddingL
+		lines = append(lines, fmt.Sprintf("%s %s%s%s %s", vertical, strings.Repeat(" ", paddingL), line, strings.Repeat(" ", paddingR), vertical))
+
+	}
+	lines = append(lines, corners[2]+strings.Repeat(horizontal, maxLineLength+2)+corners[3])
```

Comment:
```suggestion
// Global:
const horizontal = "═"
(...)
	lines := []string{messageBoxTopEdge(maxLineLength)}
	for _, line := range message {
		lines = append(lines, messageBoxLine(line, maxLineLength))
	}
	lines = append(lines, messageBoxBottomEdge(maxLineLength))
(...)
func centerText(text string, length int) string {
	return fmt.Sprintf("%*s", -length, fmt.Sprintf("%*s", (length+len(text))/2, text))
}

func messageBoxBottomEdge(length int) string {
	return fmt.Sprintf("╚%s╝", strings.Repeat(horizontal, length))
}

func messageBoxLine(text string, length int) string {
	return fmt.Sprintf("║%s║", centerText(text, length))
}

func messageBoxTopEdge(length int) string {
	return fmt.Sprintf("╔%s╗", strings.Repeat(horizontal, length))
}
```

Reply:
I'll split the functions out, but that format string makes my head hurt. I'm going to keep the explicit padding calculations. I think they are more readable.